validator_client/src/http_api/mod.rs

```diff
@@ -1108,7 +1109,7 @@ pub fn serve<T: 'static + SlotClock + Clone, E: EthSpec>(
         .and(validator_store_filter)
         .and(task_executor_filter)
         .and(log_filter.clone())
-        .and_then(|request, signer, validator_store, task_executor, log| {
+        .then(|request, signer, validator_store, task_executor, log| {
             blocking_signed_json_task(signer, move || {
                 remotekeys::delete(request, validator_store, task_executor, log)
             })
```

Comment:
There's another occurrence of `and_then` here:
https://github.com/sigp/lighthouse/blob/f77a5a0a4e7bc565bb7079998eb81c2bdd85ccf4/validator_client/src/http_api/mod.rs#L1124

Reply:
Fixed